### DISCUSSION TITLE
Different ATtiny1616 batch, different performance

### DISCUSSION THREAD

**Discussion by kinesphere** [SpenceKonde/megaTinyCore#863] [Q&A] [answered]
Hi everyone! I've been using the ATtiny1616 for a small buzzer product. A 3V CR1225 powers it and the buzzer is an SMD piezo 9018. It runs on a 2MHz internal clock and RTC timer.

I've been ordering assembled batches from JLCPCB and everything was working fine. However, I ordered a larger batch that came this week and it's completely unusable. The tone output is very quiet, and while it should be 492Hz, it gives 615Hz. So I replaced the piezo with one from the previous batch (along with all other passive components) but nothing changed, so it seems that the problem is with the ATtiny1616 chip (or the PCB). I tested reprogramming it with different settings but the result is always the same: super quiet and higher pitch in comparison to previous batches.

I imagine the higher pitch is due to the factory tuning of the crystal. But why is it so quiet? I measured the voltage output (PA4) to the piezo on the oscilloscope (2.2V) and it's the same for the new and old batches, so I think it's a current issue. 

I wonder if I received defective or maybe fake chips. Has anyone ever had a similar issue? Any suggestion?

**Comment by hmeijdam** [comment, marked as answer]
Did you test programming one of your old batch with the IDE you are currently using? That would eliminate a software issue if it then works ok.

**Reply by kinesphere** [reply]
Yes, I did. The old batch works the same as before when reprogrammed.

**Reply by hmeijdam** [reply]
I would then have the same reservations about the T1616 as you have. Having a hotair soldering station I would lift the new T1616 and solder it to the old board to get confirmation it's different.

**Reply by kinesphere** [reply]
I tried it, but the VQFN is a tough one in a small area. 

It seems I had made some confusion with my old batches (a mix of 1MHz and 2MHz), and I didn't test the previous one (2MHz) properly.

Anyway, problem was solved after rebooting my computer. The update actually fixes it, I tested in my colleage's computer (which didn't need to reboot for the update) as well. 

With megaTinyCore v. 2.6.1 and programming at 2MHz we get the problem. But with the latest (2.6.4) it works well, louder and with the correct frequency. Some SMD piezo were also not good. In a batch of 50pcs, many were quieter than usual (I guess this brand we are using is not very reliable, probably they get damaged when reflow).

Phew, what a relief. This thing was driving me nuts. :)
